src/route-recognizer.js

```diff
@@ -254,6 +263,21 @@ function parse(route, names, types, caseSensitive) {
       continue;
     }
 
+    match = segment.match(/^\{(.+?)(\?)?\}$/);
```

Comment:
I think a comment briefly describing what this is searching for would be helpful.